### PR TITLE
Pure Numpy backend

### PR DESCRIPTION
**Pull request by stavros11**
Addressing #55

### CODE REVIEW

File: tensornetwork/backend/numpy/decompositions.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+"""Tensor Decomposition Numpy Implementation."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import numpy as np
+from typing import Optional, Tuple
+
+
+def svd_decomposition(tensor: np.ndarray,
```
Comment (Thenerdstation) [inline tensornetwork/backend/numpy/decompositions.py]: Can you remove the numpy import, add a argument `np` and pass `self.np` into this function? 

Normally I wouldn't make you do this, but by doing it that way we can get JAX support for free.

Comment (stavros11) [reply tensornetwork/backend/numpy/decompositions.py]: Oops, yes, I forgot to do this in decompositions. Just make sure typing is okay.

**Comment by Thenerdstation** [comment]
Add `__init__.py`, and go ahead and do the `Tensor = Any` hack we've been doing to replace the `"np.array"` type issues.

**Comment by stavros11** [comment]
Okay, changed everything to `Tensor = Any`. Can you fix the typing of the `np` argument in `svd_decomposition` because I am not sure what that should be. Also, will we continue using `tf.test` in the tests for all backends?

**Comment by Thenerdstation** [comment]
> Okay, changed everything to Tensor = Any. Can you fix the typing of the np argument in svd_decomposition because I am not sure what that should be.

Yeah I can do that.

> Also, will we continue using tf.test in the tests for all backends?

Probably not. The only `tf.test.TestCase` thing we use is `self.AssertAllClose` and I tested replacing it with `np.testsing.assert_allclose` and it worked for TF and PyTorch. We'll just replace it with that. (pytest doesn't even require using classes).

It might cause some issues internally but we'll deal with that when we get to it.

**Comment by stavros11** [comment]
I think lint doesn't like my `__init__.py` but I think the rest is okay. I can change that to `np.testing` then (I have used this internally too).

**Comment by Thenerdstation** [comment]
Yeah, go ahead and do that in a different PR.

**Pull request merged**